# 6.1. - New York Citi Bikes Project - Ivan Dimitrov

## Importing Libraries

In [69]:
# Importing Libraries
import pandas as pd
import numpy as np
import os

## Creating a Path

In [70]:
# Creating a path
path = r'C:\Users\ivand\Desktop\6.1 - CitiBike Project Folder'

## Importing the Data Set

In [71]:
# Importing the data set
df_bikes = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'citibike.csv'))

In [72]:
# Checking the data
df_bikes

trip_id  bike_id weekday  start_hour           start_time  \
0      LnQzQk    16013     Mon          18  2013-09-09 18:18:55   
1      IL9boN    15230     Thu          18  2013-09-12 18:38:53   
2      46clGB    17942     Wed          19  2013-09-18 19:44:04   
3      v7vdFt    19683     Sat          11  2013-09-28 11:54:37   
4      VGBsb5    18024     Sat          18  2013-09-07 18:08:22   
...       ...      ...     ...         ...                  ...   
49995  Fb2bmC    17988     Sat          18  2013-09-21 18:39:20   
49996  XGB7JU    18458     Sat          17  2013-09-14 17:32:26   
49997  vFs2HS    15054     Sat          18  2013-09-07 18:08:21   
49998  xxacSt    16763     Sat          13  2013-09-07 13:25:35   
49999  4ZmicF    16794     Mon          21  2013-09-23 21:54:36   

       start_station_id       start_station_name  start_station_latitude  \
0                   523          W 38 St & 8 Ave               40.754666   
1                   257  Lispenard St & Broadway               40.719392   
2                   479          9 Ave & W 45 St               40.760193   
3                   527          E 33 St & 1 Ave               40.743156   
4                   521          8 Ave & W 31 St               40.750450   
...                 ...                      ...                     ...   
49995               479          9 Ave & W 45 St               40.760193   
49996               261     Johnson St & Gold St               40.694749   
49997               402       Broadway & E 22 St               40.740343   
49998               271   Ashland Pl & Hanson Pl               40.685282   
49999               515         W 43 St & 10 Ave               40.760094   

       start_station_longitude             end_time  end_station_id  \
0                   -73.991382  2013-09-09 18:35:28             334   
1                   -74.002472  2013-09-12 18:48:34             236   
2                   -73.991255  2013-09-18 19:50:05             513   
3                   -73.974347  2013-09-28 12:03:58             441   
4                   -73.994811  2013-09-07 18:46:38             476   
...                        ...                  ...             ...   
49995               -73.991255  2013-09-21 20:08:59             478   
49996               -73.983625  2013-09-14 17:52:17             151   
49997               -73.989551  2013-09-07 18:25:51             524   
49998               -73.978058  2013-09-07 14:14:04             373   
49999               -73.994618  2013-09-23 22:00:46             488   

                   end_station_name  end_station_latitude  \
0                   W 20 St & 7 Ave             40.742388   
1               St Marks Pl & 2 Ave             40.728419   
2                  W 56 St & 10 Ave             40.768254   
3                   E 52 St & 2 Ave             40.756014   
4                   E 31 St & 3 Ave             40.743943   
...                             ...                   ...   
49995              11 Ave & W 41 St             40.760301   
49996      Cleveland Pl & Spring St             40.721816   
49997               W 43 St & 6 Ave             40.755273   
49998  Willoughby Ave & Walworth St             40.693317   
49999               W 39 St & 9 Ave             40.756458   

       end_station_longitude  trip_duration      subscriber  birth_year  \
0                 -73.997262            993      Subscriber      1968.0   
1                 -73.987140            581      Subscriber      1983.0   
2                 -73.988639            361      Subscriber      1989.0   
3                 -73.967416            561      Subscriber      1988.0   
4                 -73.979661           2296  Non-Subscriber         NaN   
...                      ...            ...             ...         ...   
49995             -73.998842           2697  Non-Subscriber         NaN   
49996             -73.997203           1191      Subscriber      1970.0   
49997             -73.98316

## Data Checks and Cleaning

### Consistency

In [73]:
# Checking the dimensions of the data set
df_bikes.shape

(50000, 18)

In [74]:
# Looking for missing values
df_bikes.isnull().sum()

trip_id                       0
bike_id                       0
weekday                       0
start_hour                    0
start_time                    0
start_station_id              0
start_station_name            0
start_station_latitude        0
start_station_longitude       0
end_time                      0
end_station_id                0
end_station_name              0
end_station_latitude          0
end_station_longitude         0
trip_duration                 0
subscriber                    0
birth_year                 6979
gender                        0
dtype: int64

- there are (6979) missing values in the "birth_year" column

#### I don't want to drop almost 7k customers, just because they don't have birht_year information. It not only takes 14% of the whole data set, but the other parts of the information regarding those customers can be useful. So I've decided to input the NaN values with the mode of the birth_year.

In [75]:
# Checking the data set for duplicates
df_bikes_dups = df_bikes[df_bikes.duplicated()]

In [76]:
df_bikes_dups

Empty DataFrame
Columns: [trip_id, bike_id, weekday, start_hour, start_time, start_station_id, start_station_name, start_station_latitude, start_station_longitude, end_time, end_station_id, end_station_name, end_station_latitude, end_station_longitude, trip_duration, subscriber, birth_year, gender]
Index: []

- there are NO duplicate values found in the data set

In [77]:
# Checking for mixed-type values
for col in df_bikes.columns.tolist():
  weird = (df_bikes[[col]].applymap(type) != df_bikes[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_bikes[weird]) > 0:
    print (col)

- there are no Mixed-type values in the data set

### Wrangling

In [78]:
# Checking the columns and basic info
df_bikes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 18 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   trip_id                  50000 non-null  object 
 1   bike_id                  50000 non-null  int64  
 2   weekday                  50000 non-null  object 
 3   start_hour               50000 non-null  int64  
 4   start_time               50000 non-null  object 
 5   start_station_id         50000 non-null  int64  
 6   start_station_name       50000 non-null  object 
 7   start_station_latitude   50000 non-null  float64
 8   start_station_longitude  50000 non-null  float64
 9   end_time                 50000 non-null  object 
 10  end_station_id           50000 non-null  int64  
 11  end_station_name         50000 non-null  object 
 12  end_station_latitude     50000 non-null  float64
 13  end_station_longitude    50000 non-null  float64
 14  trip_duration         

In [79]:
# Dropping columns I don't need for the potential questions and analysis
df_bikes_clean = df_bikes.drop(columns = ['trip_id','bike_id','start_station_id','end_station_id','start_hour'])

In [80]:
# Checking the cleaned data
df_bikes_clean

weekday           start_time       start_station_name  \
0         Mon  2013-09-09 18:18:55          W 38 St & 8 Ave   
1         Thu  2013-09-12 18:38:53  Lispenard St & Broadway   
2         Wed  2013-09-18 19:44:04          9 Ave & W 45 St   
3         Sat  2013-09-28 11:54:37          E 33 St & 1 Ave   
4         Sat  2013-09-07 18:08:22          8 Ave & W 31 St   
...       ...                  ...                      ...   
49995     Sat  2013-09-21 18:39:20          9 Ave & W 45 St   
49996     Sat  2013-09-14 17:32:26     Johnson St & Gold St   
49997     Sat  2013-09-07 18:08:21       Broadway & E 22 St   
49998     Sat  2013-09-07 13:25:35   Ashland Pl & Hanson Pl   
49999     Mon  2013-09-23 21:54:36         W 43 St & 10 Ave   

       start_station_latitude  start_station_longitude             end_time  \
0                   40.754666               -73.991382  2013-09-09 18:35:28   
1                   40.719392               -74.002472  2013-09-12 18:48:34   
2                   40.760193               -73.991255  2013-09-18 19:50:05   
3                   40.743156               -73.974347  2013-09-28 12:03:58   
4                   40.750450               -73.994811  2013-09-07 18:46:38   
...                       ...                      ...                  ...   
49995               40.760193               -73.991255  2013-09-21 20:08:59   
49996               40.694749               -73.983625  2013-09-14 17:52:17   
49997               40.740343               -73.989551  2013-09-07 18:25:51   
49998               40.685282               -73.978058  2013-09-07 14:14:04   
49999               40.760094               -73.994618  2013-09-23 22:00:46   

                   end_station_name  end_station_latitude  \
0                   W 20 St & 7 Ave             40.742388   
1               St Marks Pl & 2 Ave             40.728419   
2                  W 56 St & 10 Ave             40.768254   
3                   E 52 St & 2 Ave             40.756014   
4                   E 31 St & 3 Ave             40.743943   
...                             ...                   ...   
49995              11 Ave & W 41 St             40.760301   
49996      Cleveland Pl & Spring St             40.721816   
49997               W 43 St & 6 Ave             40.755273   
49998  Willoughby Ave & Walworth St             40.693317   
49999               W 39 St & 9 Ave             40.756458   

       end_station_longitude  trip_duration      subscriber  birth_year  \
0                 -73.997262            993      Subscriber      1968.0   
1                 -73.987140            581      Subscriber      1983.0   
2                 -73.988639            361      Subscriber      1989.0   
3                 -73.967416            561      Subscriber      1988.0   
4                 -73.979661           2296  Non-Subscriber         NaN   
...                      ...            ...             ...         ...   
49995             -73.998842           2697  Non-Subscriber         NaN   
49996             -73.997203           1191      Subscriber      1970.0   
49997             -73.983169           1050      Subscriber      1961.0   
49998             -73.953820           2697  Non-Subscriber         NaN   
49999             -73.993722            370      Subscriber      1980.0   

       gender  
0           2  
1           1  
2           1  
3           2  
4           0  
...       ...  
49995       0  
49996       2  
49997       2  
49998       0  
49999       2  

[50000 rows x 13 columns]

In [81]:
# Checking data info after dropped columns
df_bikes_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   weekday                  50000 non-null  object 
 1   start_time               50000 non-null  object 
 2   start_station_name       50000 non-null  object 
 3   start_station_latitude   50000 non-null  float64
 4   start_station_longitude  50000 non-null  float64
 5   end_time                 50000 non-null  object 
 6   end_station_name         50000 non-null  object 
 7   end_station_latitude     50000 non-null  float64
 8   end_station_longitude    50000 non-null  float64
 9   trip_duration            50000 non-null  int64  
 10  subscriber               50000 non-null  object 
 11  birth_year               43021 non-null  float64
 12  gender                   50000 non-null  int64  
dtypes: float64(5), int64(2), object(6)
memory usage: 5.0+ MB


In [82]:
# Changing the type of data to lower the memory usage of the file
df_bikes_clean['trip_duration']=df_bikes_clean['trip_duration'].astype('int16')

In [83]:
# Changing gender to a string value
df_bikes_clean['gender']=df_bikes_clean['gender'].astype('str')

In [84]:
# Checking data info after type change
df_bikes_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   weekday                  50000 non-null  object 
 1   start_time               50000 non-null  object 
 2   start_station_name       50000 non-null  object 
 3   start_station_latitude   50000 non-null  float64
 4   start_station_longitude  50000 non-null  float64
 5   end_time                 50000 non-null  object 
 6   end_station_name         50000 non-null  object 
 7   end_station_latitude     50000 non-null  float64
 8   end_station_longitude    50000 non-null  float64
 9   trip_duration            50000 non-null  int16  
 10  subscriber               50000 non-null  object 
 11  birth_year               43021 non-null  float64
 12  gender                   50000 non-null  object 
dtypes: float64(5), int16(1), object(7)
memory usage: 4.7+ MB


In [85]:
# Changing the "gender" column to "int8" so we can perform the transformation of (0,1,2) to (unknown,male,female)
df_bikes_clean['gender']=df_bikes_clean['gender'].astype('int8')

In [86]:
# Changing the gender variables to actual genders instead of numbers
df_bikes_clean['gender']=df_bikes_clean['gender'].replace([0,1,2],['Unknown','Male','Female'])

In [87]:
# Reverting "gender" column back to a string
df_bikes_clean['gender']=df_bikes_clean['gender'].astype('str')

In [88]:
# Checking the result
df_bikes_clean.head()

weekday           start_time       start_station_name  \
0     Mon  2013-09-09 18:18:55          W 38 St & 8 Ave   
1     Thu  2013-09-12 18:38:53  Lispenard St & Broadway   
2     Wed  2013-09-18 19:44:04          9 Ave & W 45 St   
3     Sat  2013-09-28 11:54:37          E 33 St & 1 Ave   
4     Sat  2013-09-07 18:08:22          8 Ave & W 31 St   

   start_station_latitude  start_station_longitude             end_time  \
0               40.754666               -73.991382  2013-09-09 18:35:28   
1               40.719392               -74.002472  2013-09-12 18:48:34   
2               40.760193               -73.991255  2013-09-18 19:50:05   
3               40.743156               -73.974347  2013-09-28 12:03:58   
4               40.750450               -73.994811  2013-09-07 18:46:38   

      end_station_name  end_station_latitude  end_station_longitude  \
0      W 20 St & 7 Ave             40.742388             -73.997262   
1  St Marks Pl & 2 Ave             40.728419             -73.987140   
2     W 56 St & 10 Ave             40.768254             -73.988639   
3      E 52 St & 2 Ave             40.756014             -73.967416   
4      E 31 St & 3 Ave             40.743943             -73.979661   

   trip_duration      subscriber  birth_year   gender  
0            993      Subscriber      1968.0   Female  
1            581      Subscriber      1983.0     Male  
2            361      Subscriber      1989.0     Male  
3            561      Subscriber      1988.0   Female  
4           2296  Non-Subscriber         NaN  Unknown

In [89]:
# Showing the descriptive stats of the data set
df_bikes_clean.describe()

start_station_latitude  start_station_longitude  end_station_latitude  \
count            50000.000000             50000.000000          50000.000000   
mean                40.734170               -73.991109             40.733859   
std                  0.019911                 0.012555              0.019885   
min                 40.680342               -74.017134             40.680342   
25%                 40.720196               -74.000271             40.720196   
50%                 40.735877               -73.990765             40.735354   
75%                 40.750020               -73.981923             40.749013   
max                 40.770513               -73.950048             40.770513   

       end_station_longitude  trip_duration    birth_year  
count           50000.000000   50000.000000  43021.000000  
mean              -73.991351     838.982900   1975.627786  
std                 0.012569     573.663997     11.089001  
min               -74.017134      60.000000   1899.000000  
25%               -74.001547     417.000000   1968.000000  
50%               -73.991218     672.000000   1978.000000  
75%               -73.982050    1112.000000   1984.000000  
max               -73.950048    2697.000000   1997.000000

In [90]:
# Replacing the NaN values inside column "birth_year" with the Mode of the data
df_bikes_new = df_bikes_clean.fillna("1978.000000")

In [91]:
# Checking the data after the changes
df_bikes_new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   weekday                  50000 non-null  object 
 1   start_time               50000 non-null  object 
 2   start_station_name       50000 non-null  object 
 3   start_station_latitude   50000 non-null  float64
 4   start_station_longitude  50000 non-null  float64
 5   end_time                 50000 non-null  object 
 6   end_station_name         50000 non-null  object 
 7   end_station_latitude     50000 non-null  float64
 8   end_station_longitude    50000 non-null  float64
 9   trip_duration            50000 non-null  int16  
 10  subscriber               50000 non-null  object 
 11  birth_year               50000 non-null  object 
 12  gender                   50000 non-null  object 
dtypes: float64(4), int16(1), object(8)
memory usage: 4.7+ MB


In [92]:
# Changing "birth_year" column to a float so we can input the Mode value in the NaN place
df_bikes_new['birth_year']=df_bikes_new['birth_year'].astype('float64')

In [93]:
# Checking the data
df_bikes_new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   weekday                  50000 non-null  object 
 1   start_time               50000 non-null  object 
 2   start_station_name       50000 non-null  object 
 3   start_station_latitude   50000 non-null  float64
 4   start_station_longitude  50000 non-null  float64
 5   end_time                 50000 non-null  object 
 6   end_station_name         50000 non-null  object 
 7   end_station_latitude     50000 non-null  float64
 8   end_station_longitude    50000 non-null  float64
 9   trip_duration            50000 non-null  int16  
 10  subscriber               50000 non-null  object 
 11  birth_year               50000 non-null  float64
 12  gender                   50000 non-null  object 
dtypes: float64(5), int16(1), object(7)
memory usage: 4.7+ MB


In [94]:
# Visualizing the change
df_bikes_new.describe()

start_station_latitude  start_station_longitude  end_station_latitude  \
count            50000.000000             50000.000000          50000.000000   
mean                40.734170               -73.991109             40.733859   
std                  0.019911                 0.012555              0.019885   
min                 40.680342               -74.017134             40.680342   
25%                 40.720196               -74.000271             40.720196   
50%                 40.735877               -73.990765             40.735354   
75%                 40.750020               -73.981923             40.749013   
max                 40.770513               -73.950048             40.770513   

       end_station_longitude  trip_duration    birth_year  
count           50000.000000   50000.000000  50000.000000  
mean              -73.991351     838.982900   1975.958900  
std                 0.012569     573.663997     10.318811  
min               -74.017134      60.000000   1899.000000  
25%               -74.001547     417.000000   1970.000000  
50%               -73.991218     672.000000   1978.000000  
75%               -73.982050    1112.000000   1983.000000  
max               -73.950048    2697.000000   1997.000000

In [95]:
# Reverting back the "birth_year" column to a string in order to see the years as whole numbers instead of continuous
df_bikes_new['birth_year'] = df_bikes_new['birth_year'].astype('int16')

In [96]:
# Checking the data after all the changes
df_bikes_new

weekday           start_time       start_station_name  \
0         Mon  2013-09-09 18:18:55          W 38 St & 8 Ave   
1         Thu  2013-09-12 18:38:53  Lispenard St & Broadway   
2         Wed  2013-09-18 19:44:04          9 Ave & W 45 St   
3         Sat  2013-09-28 11:54:37          E 33 St & 1 Ave   
4         Sat  2013-09-07 18:08:22          8 Ave & W 31 St   
...       ...                  ...                      ...   
49995     Sat  2013-09-21 18:39:20          9 Ave & W 45 St   
49996     Sat  2013-09-14 17:32:26     Johnson St & Gold St   
49997     Sat  2013-09-07 18:08:21       Broadway & E 22 St   
49998     Sat  2013-09-07 13:25:35   Ashland Pl & Hanson Pl   
49999     Mon  2013-09-23 21:54:36         W 43 St & 10 Ave   

       start_station_latitude  start_station_longitude             end_time  \
0                   40.754666               -73.991382  2013-09-09 18:35:28   
1                   40.719392               -74.002472  2013-09-12 18:48:34   
2                   40.760193               -73.991255  2013-09-18 19:50:05   
3                   40.743156               -73.974347  2013-09-28 12:03:58   
4                   40.750450               -73.994811  2013-09-07 18:46:38   
...                       ...                      ...                  ...   
49995               40.760193               -73.991255  2013-09-21 20:08:59   
49996               40.694749               -73.983625  2013-09-14 17:52:17   
49997               40.740343               -73.989551  2013-09-07 18:25:51   
49998               40.685282               -73.978058  2013-09-07 14:14:04   
49999               40.760094               -73.994618  2013-09-23 22:00:46   

                   end_station_name  end_station_latitude  \
0                   W 20 St & 7 Ave             40.742388   
1               St Marks Pl & 2 Ave             40.728419   
2                  W 56 St & 10 Ave             40.768254   
3                   E 52 St & 2 Ave             40.756014   
4                   E 31 St & 3 Ave             40.743943   
...                             ...                   ...   
49995              11 Ave & W 41 St             40.760301   
49996      Cleveland Pl & Spring St             40.721816   
49997               W 43 St & 6 Ave             40.755273   
49998  Willoughby Ave & Walworth St             40.693317   
49999               W 39 St & 9 Ave             40.756458   

       end_station_longitude  trip_duration      subscriber  birth_year  \
0                 -73.997262            993      Subscriber        1968   
1                 -73.987140            581      Subscriber        1983   
2                 -73.988639            361      Subscriber        1989   
3                 -73.967416            561      Subscriber        1988   
4                 -73.979661           2296  Non-Subscriber        1978   
...                      ...            ...             ...         ...   
49995             -73.998842           2697  Non-Subscriber        1978   
49996             -73.997203           1191      Subscriber        1970   
49997             -73.983169           1050      Subscriber        1961   
49998             -73.953820           2697  Non-Subscriber        1978   
49999             -73.993722            370      Subscriber        1980   

        gender  
0       Female  
1         Male  
2         Male  
3       Female  
4      Unknown  
...        ...  
49995  Unknown  
49996   Female  
49997   Female  
49998  Unknown  
49999   Female  

[50000 rows x 13 columns]

In [97]:
# Renaming some columns for ease of use
df_bikes_new.rename(columns = {'weekday':'day_of_week'}, inplace = True)
df_bikes_new.rename(columns = {'start_time':'start_of_ride'}, inplace = True)
df_bikes_new.rename(columns = {'end_time':'end_of_ride'}, inplace = True)
df_bikes_new.rename(columns = {'trip_duration':'trip_duration_in_minutes'}, inplace = True)
df_bikes_new.rename(columns = {'birth_year':'year_of_birth'}, inplace = True)

In [98]:
# Checking the new dataset
df_bikes_new

day_of_week        start_of_ride       start_station_name  \
0             Mon  2013-09-09 18:18:55          W 38 St & 8 Ave   
1             Thu  2013-09-12 18:38:53  Lispenard St & Broadway   
2             Wed  2013-09-18 19:44:04          9 Ave & W 45 St   
3             Sat  2013-09-28 11:54:37          E 33 St & 1 Ave   
4             Sat  2013-09-07 18:08:22          8 Ave & W 31 St   
...           ...                  ...                      ...   
49995         Sat  2013-09-21 18:39:20          9 Ave & W 45 St   
49996         Sat  2013-09-14 17:32:26     Johnson St & Gold St   
49997         Sat  2013-09-07 18:08:21       Broadway & E 22 St   
49998         Sat  2013-09-07 13:25:35   Ashland Pl & Hanson Pl   
49999         Mon  2013-09-23 21:54:36         W 43 St & 10 Ave   

       start_station_latitude  start_station_longitude          end_of_ride  \
0                   40.754666               -73.991382  2013-09-09 18:35:28   
1                   40.719392               -74.002472  2013-09-12 18:48:34   
2                   40.760193               -73.991255  2013-09-18 19:50:05   
3                   40.743156               -73.974347  2013-09-28 12:03:58   
4                   40.750450               -73.994811  2013-09-07 18:46:38   
...                       ...                      ...                  ...   
49995               40.760193               -73.991255  2013-09-21 20:08:59   
49996               40.694749               -73.983625  2013-09-14 17:52:17   
49997               40.740343               -73.989551  2013-09-07 18:25:51   
49998               40.685282               -73.978058  2013-09-07 14:14:04   
49999               40.760094               -73.994618  2013-09-23 22:00:46   

                   end_station_name  end_station_latitude  \
0                   W 20 St & 7 Ave             40.742388   
1               St Marks Pl & 2 Ave             40.728419   
2                  W 56 St & 10 Ave             40.768254   
3                   E 52 St & 2 Ave             40.756014   
4                   E 31 St & 3 Ave             40.743943   
...                             ...                   ...   
49995              11 Ave & W 41 St             40.760301   
49996      Cleveland Pl & Spring St             40.721816   
49997               W 43 St & 6 Ave             40.755273   
49998  Willoughby Ave & Walworth St             40.693317   
49999               W 39 St & 9 Ave             40.756458   

       end_station_longitude  trip_duration_in_minutes      subscriber  \
0                 -73.997262                       993      Subscriber   
1                 -73.987140                       581      Subscriber   
2                 -73.988639                       361      Subscriber   
3                 -73.967416                       561      Subscriber   
4                 -73.979661                      2296  Non-Subscriber   
...                      ...                       ...             ...   
49995             -73.998842                      2697  Non-Subscriber   
49996             -73.997203                      1191      Subscriber   
49997             -73.983169                      1050      Subscriber   
49998             -73.953820                      2697  Non-Subscriber   
49999             -73.993722                       370      Subscriber   

       year_of_birth   gender  
0               1968   Female  
1               1983     Male  
2               1989     Male  
3               1988   Female  
4               1978  Unknown  
...              ...      ...  
49995           1978  Unknown  
49996           1970   Female  
49997           1961   Female  
49998           1978  Unknown  
49999           1980   Female  

[50000 rows x 13 columns]

In [99]:
# Removing data for people born before 1933, because I do not consider bike riding data for people aged above 80 years old as relevant
df_bikes_latest = df_bikes_new[df_bikes_new['year_of_birth'] >= 1938]

In [100]:
# Checking the filtered data
df_bikes_latest.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 49927 entries, 0 to 49999
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   day_of_week               49927 non-null  object 
 1   start_of_ride             49927 non-null  object 
 2   start_station_name        49927 non-null  object 
 3   start_station_latitude    49927 non-null  float64
 4   start_station_longitude   49927 non-null  float64
 5   end_of_ride               49927 non-null  object 
 6   end_station_name          49927 non-null  object 
 7   end_station_latitude      49927 non-null  float64
 8   end_station_longitude     49927 non-null  float64
 9   trip_duration_in_minutes  49927 non-null  int16  
 10  subscriber                49927 non-null  object 
 11  year_of_birth             49927 non-null  int16  
 12  gender                    49927 non-null  object 
dtypes: float64(4), int16(2), object(7)
memory usage: 4.8+ MB


## Exporting the Cleaned data as CSV file

In [101]:
df_bikes_latest.to_csv(os.path.join(path, '02 Data','Prepared Data', 'CitiBike_Cleaned.csv'))